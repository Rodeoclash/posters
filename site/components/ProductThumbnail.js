import Link from 'next/link'

const ProductThumbnail = ({ product }) => (
  <dl>
    <dt>Image</dt>
    <dd>
      <img src={product.src} />
    </dd>
    <dt>Title</dt>
    <dd>
      {product.title}
    </dd>
    <dt>Description</dt>
    <dd>
      {product.description}
    </dd>
    <dt>Link to product page</dt>
    <dd>
      <Link href="/posters/[slug]" as={'/posters/'+product.handle}>
        <a>Link to product page</a>
      </Link>
    </dd>
  </dl>
);

export default ProductThumbnail;
