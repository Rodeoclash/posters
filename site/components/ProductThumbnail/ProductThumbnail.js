import Link from "next/link";
import ProductAddToCart from "../ProductAddToCart";
import ProductImage from "../ProductImage";
import ProductInCart from "../ProductInCart";
import ProductLink from "../ProductLink";

import styles from "./ProductThumbnail.module.css";

const ProductThumbnail = ({ product }) => (
  <div className={styles.root}>
    <ProductImage product={product} />
    <h3>{product.title}</h3>
    <div>
      <div>
        <ProductLink product={product} />
      </div>
      <div>
        <ProductAddToCart product={product} />
        <ProductInCart product={product} />
      </div>
    </div>
  </div>
);

export default ProductThumbnail;
