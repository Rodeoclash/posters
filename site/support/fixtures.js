// Products as returned by Shopify
export const shopifyProducts = [
  {
    description: "description 1",
    handle: "handle-1",
    id: "id 1",
    images: [{ src: "src 1" }],
    title: "title 1",
  },
  {
    description: "description 2",
    handle: "handle-2",
    id: "id 2",
    images: [{ src: "src 2" }],
    title: "title 2",
  },
];

// Products as given to frontend (transformed from shopifyProducts above)
export const products = [
  {
    description: "description 1",
    handle: "handle-1",
    id: "id 1",
    src: "src 1",
    title: "title 1",
  },
  {
    description: "description 2",
    handle: "handle-2",
    id: "id 2",
    src: "src 2",
    title: "title 2",
  },
];
