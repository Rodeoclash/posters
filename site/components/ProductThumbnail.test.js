import React from "react";
import renderer from "react-test-renderer";
import { products } from "../support/fixtures";

import ProductThumbnail from "./ProductThumbnail";

it("renders correctly", () => {
  const tree = renderer
    .create(<ProductThumbnail product={products[0]} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
