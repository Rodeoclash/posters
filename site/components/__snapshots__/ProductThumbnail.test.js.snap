// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<dl>
  <dt>
    Image
  </dt>
  <dd>
    <img
      src="src 1"
    />
  </dd>
  <dt>
    Title
  </dt>
  <dd>
    title 1
  </dd>
  <dt>
    Description
  </dt>
  <dd>
    description 1
  </dd>
</dl>
`;
